### PR TITLE
feat: upgrading Mender Server: general gudelines.

### PR DESCRIPTION
**Pull request by merlin-northern**
This is as we agreed during today's meeting, the first out of 3 PRs.

Changelog: Title
Ticket: MEN-6219

### CODE REVIEW

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+It is the first step in the upgrade preparation. Ideally, you should have a backup
+that you are sure is valid, that is: you have means to restore it to some environment
+and verify your setup works.
+
+#### MongoDB
+
+Be sure you have an up-to-date, working, and complete backup of your Mender MongoDB server
+in a format identical to the one generated by `mongodump` command.
+
+### Artifacts storage (S3, or compatible)
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/02.Upgrading-from-previous-versions/docs.md:39]:
```suggestion
**Artifacts storage (Aws S3 or Azure Blob storage)**
```

Would keep to those we support.
Difference between ### and #### doesn't render as a subchapter

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+## Prerequisites
+
+### Taking a full backup
+
+In the following chapters the backup will be always the first item on the agenda.
+It is the first step in the upgrade preparation. Ideally, you should have a backup
+that you are sure is valid, that is: you have means to restore it to some environment
+and verify your setup works.
+
+#### MongoDB
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/02.Upgrading-from-previous-versions/docs.md:34]:
```suggestion
**MongoDB**
```

Difference between ### and #### doesn't render as a subchapter.

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+Be sure you have an up-to-date, working, and complete backup of your Mender MongoDB server
+in a format identical to the one generated by `mongodump` command.
+
+### Artifacts storage (S3, or compatible)
+
+Be sure you either enable versioning on your S3 buckets or otherwise backup the artifact
+storage. During the upgrade procedure it is virtually impossible that something can go wrong
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/02.Upgrading-from-previous-versions/docs.md:42]:
```suggestion
storage. During the upgrade procedure it is highly unlikely that something can go wrong
```

**Review by TheMeaningfulEngineer — COMMENTED** [review]
Looks good, just some minor formatting changes

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md (hunk 3 above)
Comment (merlin-northern) [reply 07.Server-installation/02.Upgrading-from-previous-versions/docs.md]: fixed.

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md (hunk 2 above)
Comment (merlin-northern) [reply 07.Server-installation/02.Upgrading-from-previous-versions/docs.md]: strange! I really thought there used to be a difference :) okey, what I did: I removed the `## Prerequisites` which we did not need, and used `##` and `###` only. is that okey?

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md (hunk 1 above)
Comment (merlin-northern) [reply 07.Server-installation/02.Upgrading-from-previous-versions/docs.md]: strange! I really thought there used to be a difference :) okey, what I did: I removed the `## Prerequisites` which we did not need, and used `##` and `###` only. is that okey?

Comment (TheMeaningfulEngineer) [reply 07.Server-installation/02.Upgrading-from-previous-versions/docs.md]: Works, tnx

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+* verify that the backup works
+* be sure that new versions properly upgrade the database
+
+## Maintenance window
+
+Be prepared that in some complicated upgrades there may be a period when the Mender Server
+is not available, or otherwise presents lack of proper service, or incomplete data to the clients.
+Exercising our imagination we can come up with a scenario when during a certain upgrade a flood
+of request will lead to data inconsistencies. To mitigate that risk it is better to prepare
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/02.Upgrading-from-previous-versions/docs.md:58]:
```suggestion
of requests will lead to data inconsistencies. To mitigate that risk it is better to prepare
```

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+Be sure you either enable versioning on your S3 buckets or otherwise backup the artifact
+storage. During the upgrade procedure it is highly unlikely that something can go wrong
+with the blob storage, however we recommend being extra careful, and especially so during
+the restore to staging (see the next section).
+
+If you are using an S3 compatible storage other than the above-mentioned, be sure that
+you have an equivalent backup or versioning strategy.
+
+## Deploy to staging environment
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/02.Upgrading-from-previous-versions/docs.md:44]:
```suggestion
## Deploy a to staging environment
```

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+Upgrading the production Mender Server installations can be a complex task.
+It is supported, and encouraged, but there is a number of items to take
+into account:
+
+* upgrade between the adjacent versions, i.e.: if you want to upgrade to 3.5,
+ you should first bring your server to version 3.4.
+* always have a full backup
+* read this document
+* perform a test upgrade using a copy of your production database
+* prepare for maintenance window
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/02.Upgrading-from-previous-versions/docs.md:17]:
```suggestion
* prepare a maintenance window
```

I'm not sure

**Review by TheMeaningfulEngineer — APPROVED** [review]

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md (hunk 6 above)
Comment (merlin-northern) [reply 07.Server-installation/02.Upgrading-from-previous-versions/docs.md]: lets have it.

File: 07.Server-installation/02.Upgrading-from-previous-versions/docs.md (hunk 5 above)
Comment (merlin-northern) [reply 07.Server-installation/02.Upgrading-from-previous-versions/docs.md:44]: lets not have it.

**Pull request merged**